### PR TITLE
Converted help center provider to TypeScript

### LINKED ISSUE #12664: TypeScript: Convert helpCenter provider

### PR DESCRIPTION
**Pull request by barklund**
## Context

This converts the help center provider and some related dependencies to TypeScript

## Todo

- [x] Fix error around missing types from `design-system` package
- [x] Check tests?

## User-facing changes

_None_

## Testing Instructions

- [x] This is a non-user-facing change and requires no QA

## Checklist

- [x] This PR addresses an existing issue and I have linked this PR to it in ZenHub
- [x] I have tested this code to the best of my abilities
- [x] I have verified accessibility to the best of my abilities ([docs](https://github.com/googleforcreators/web-stories-wp/blob/main/docs/accessibility-testing.md))
- [x] I have verified i18n and l10n (translation, right-to-left layout) to the best of my abilities
- [x] This code is covered by automated tests (unit, integration, and/or e2e) to verify it works as intended ([docs](https://github.com/googleforcreators/web-stories-wp/tree/main/docs#testing))
- [x] I have added documentation where necessary
- [x] I have added a matching `Type: XYZ` label to the PR

---

Fixes #12664

### CODE REVIEW

File: packages/story-editor/src/app/currentUser/currentUserProvider.tsx [hunk 1]
```diff
@@ -24,15 +24,10 @@ import type { PropsWithChildren } from 'react';
...
-  CurrentUserState,
-  UpdateCurrentUserProps,
-} from '../../types/currentUserProvider';
+import type { CurrentUserState } from '../../types/currentUserProvider';
 import Context from './context';
 
-function CurrentUserProvider({
-  children,
-}: PropsWithChildren<Record<string, never>>) {
+function CurrentUserProvider({ children }: PropsWithChildren<unknown>) {
```
Comment (miina) [inline packages/story-editor/src/app/currentUser/currentUserProvider.tsx:30]: Why not `Record<string, never>` as it was before?

**Review by miina — APPROVED** [review]
LGTM (pending the tests passing)

File: packages/story-editor/src/app/currentUser/currentUserProvider.tsx (hunk 1 above)
Comment (barklund) [reply packages/story-editor/src/app/currentUser/currentUserProvider.tsx:30]: It caused errors when actually using it in a typed component. This is the proper way to type it (and the default in React 18 where the type is updated to `PropsWithChildren<T = unknown>`).

Comment (miina) [reply packages/story-editor/src/app/currentUser/currentUserProvider.tsx:30]: Ok, good to know!

**Pull request merged**